src/models/instance.ts

```diff
@@ -42,42 +46,47 @@ export default class Instance {
     }
 
     public setId(id: string): Instance {
-        return new Instance({...this.data, id});
+        return new Instance({ ...this.data, id });
     }
 
     public get id(): string {
         return this.data.id;
     }
 
     public setUrl(url: string): Instance {
-        return new Instance({...this.data, url});
+        return new Instance({ ...this.data, url });
     }
 
     public get url(): string {
         return this.data.url;
     }
 
     public setUsername(username: string): Instance {
-        return new Instance({...this.data, username});
+        return new Instance({ ...this.data, username });
     }
 
     public get username(): string {
         return this.data.username;
     }
 
     public setPassword(password: string): Instance {
-        return new Instance({...this.data, password});
+        return new Instance({ ...this.data, password });
     }
 
     public get password(): string {
         return this.data.password;
     }
 
     public setDescription(description: string): Instance {
-        return new Instance({...this.data, description});
+        return new Instance({ ...this.data, description });
     }
 
     public get description(): string {
         return this.data.description ? this.data.description : "";
     }
+
+    public async validateInstanceName(d2: D2): Promise<Response> {
+        const { url, username } = this.data;
+        return validateInstanceName(d2, url, username);
```

Comment:
Problem: this works on a create, but on an update, it will say the combination already exists, when in fact it's the same model we are trying to update.

Reply:
just a question, should Edit permit all fields to be altered?

Reply:
As we discussed, let's add an autogenerated ID field to the instances, it will be much easier to update things. Probably `props.originalInstance` could be then removed.